site/mex_invenio/scripts/s3_manager.py

```diff
@@ -0,0 +1,172 @@
+from datetime import datetime
+from click.testing import CliRunner
+
+# Configure logging
+log_file_path = 's3_manager.log'
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+file_handler = logging.FileHandler(log_file_path)
+file_handler.setLevel(logging.INFO)
+formatter = logging.Formatter('%(asctime)s - %(levelname)s - (line: %(lineno)d) - %(message)s')
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+def load_config():
+    config = {}
+
+    fileFound = load_dotenv()
+
+    if fileFound:
+        config = {
+            "bucket": os.getenv("bucket"),
+            "aws_access_key": os.getenv("aws_access_key"),
+            "aws_secret_key": os.getenv("aws_secret_key"),
+            "region": os.getenv("region", "eu-central-1"),
+            "email": os.getenv("email"),
+        }
+
+    return config
+
+def get_latest_file(s3_client, bucket_name):
+    try:
+        response = s3_client.list_objects_v2(Bucket=bucket_name)
+        if "Contents" not in response:
+            logger.info("No files found in the bucket.")
+            return None
+        latest_file = max(response["Contents"], key=lambda obj: obj["LastModified"])
+        return latest_file["Key"]
+    except Exception as e:
+        logger.error(f"Error fetching latest file: {e}")
+        return None
+
+def download_file(s3_client, bucket_name, file_key, payload_folder):
+    try:
+        os.makedirs(payload_folder, exist_ok=True)
+        local_filename = os.path.join(payload_folder, os.path.basename(file_key))
+        s3_client.download_file(bucket_name, file_key, local_filename)
+        return local_filename
+    except Exception as e:
+        logger.error(f"Error downloading file: {e}")
+        return None
+
+def get_latest_existing_file(payload_folder):
+    """Fetches the most recent file in the payload folder."""
+    if not os.path.exists(payload_folder):
+        os.makedirs(payload_folder)
+        return None
+    
+    files = sorted(
+        [os.path.join(payload_folder, f) for f in os.listdir(payload_folder) if os.path.isfile(os.path.join(payload_folder, f))],
+        key=os.path.getmtime,  # Sort by last modified time
+    )
+    
+    return files[-1] if files else None
+
+def check_last_download(existing_file, new_file):
+    """Compares files and deletes the new file if it's the same."""
+    if existing_file and os.path.exists(existing_file) and filecmp.cmp(existing_file, new_file, shallow=False):
+        logger.info("No new content found. File is exactly the same as before.")
+        os.remove(new_file)  # Remove duplicate file
+        return True
+    return False
+
+def rename_and_keep_latest_file(existing_file, new_file, payload_folder, check_comparison):
+    """Handles file retention based on checkLastDownload flag."""
+    if check_comparison == "yes" and check_last_download(existing_file, new_file):
+        return None  # New file is identical, so discard it
```

Comment:
Does this mean we don't have a last modified of when it last 'succeeded' on the files (even with no effect)?

Reply:
Discussed with @J4bbi and we've decided what you've done is best - the timestamped file will inform you when the last version of the data is from